python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1756,31 +1756,21 @@ def get_latest_asset_partition_materialization_attempts_without_materializations
             .alias("materialization_events")
         )
 
-        query = (
-            db.select(
-                [
-                    materialization_planned_events.c.partition,
-                    materialization_planned_events.c.run_id,
-                    materialization_planned_events.c.id,
-                ]
-            )
-            .select_from(
-                materialization_planned_events.join(
-                    materialization_events,
-                    db.and_(
-                        materialization_planned_events.c.partition
-                        == materialization_events.c.partition,
-                        materialization_planned_events.c.run_id == materialization_events.c.run_id,
-                    ),
-                    isouter=True,
-                )
-            )
-            .where(materialization_events.c.run_id == None)  # noqa: E711
-        )
-
         with self.index_connection() as conn:
-            rows = conn.execute(query).fetchall()
-            return {row["partition"]: (row["run_id"], row["id"]) for row in rows}
+            materialization_planned_rows = conn.execute(materialization_planned_events).fetchall()
+            materialization_rows = conn.execute(materialization_events).fetchall()
+
+        materialization_planned_rows_by_partition = {
+            row["partition"]: (row["run_id"], row["id"]) for row in materialization_planned_rows
+        }
+        for row in materialization_rows:
+            if (
+                row["partition"] in materialization_planned_rows_by_partition
+                and materialization_planned_rows_by_partition[row["partition"]][0] == row["run_id"]
+            ):
+                materialization_planned_rows_by_partition.pop(row["partition"], None)
```

Comment:
nit, the None isn't required I think. Missed that in cloud but doesn't really matter